aerogear/aerogear-unifiedpush-server
# Keycloak 3.x update

WIP - updates to Keycloak 3

Addresses:
* [AGPUSH-2043](https://issues.jboss.org/browse/AGPUSH-2043) 
* [AGPUSH-2117](https://issues.jboss.org/browse/AGPUSH-2117) (partially)

## How to test
- Configure/run Keycloak
  - Easiest is: run the `docker-compose/development/development-env.yaml` docker-compose file
  - check the IP address of the keycloak instance (e.g. via `docker inspect`)
- Run the AG-PUSH WAR file
  - `mvn clean install`
  - Install/unzip WF 11 and `export` its directory to `WILDFLY_HOME`
  - start the WF server with the following system properties: `-Dups.realm.name=aerogear -Dups.auth.server.url=http://IP_ADDRESS_OF_KEYCLOAK:8080/auth`
  - Inside the `databases` folder, run the `prepare_clean_wf11.sh` script 
  - Finally copy the newly build `ag-push.war` file to WF's deployment folder

NOTE: The docker-compose file for Keycloak, assumes that your WildFly (the one for UPS) is on `localhost` and `8080`!

## Expected:
* Login is redirected to the `IP OF KEYCLOAK`
* After successful UPS login, the UI comes up as normal
